api/server/backend/build/backend.go

```diff
@@ -25,10 +25,17 @@ type Backend struct {
 	imageComponent ImageComponent
 }
 
+type monolithBackend interface {
+	ImageComponent
+	builder.Backend
+	containerComponent
+}
+
 // NewBackend creates a new build backend from components
-func NewBackend(components ImageComponent, builderBackend builder.Backend) *Backend {
-	manager := dockerfile.NewBuildManager(builderBackend)
-	return &Backend{imageComponent: components, manager: manager}
+func NewBackend(monolith monolithBackend) *Backend {
+	containerBackendFunc := func() builder.ContainerBackend { return NewContainerBackend(monolith) }
+	manager := dockerfile.NewBuildManager(monolith, containerBackendFunc)
```

Comment:
This callback logic appears to exists because of the `tmpContainers` as well. I don't think there is a need for these objects to be stateful. It is dangerous to make such assumptions as 2 build jobs should be able to share each other's resources. ContainerBackend is a dependency on builder, not on the build definition.